File: Server/src/main/content/global/skill/production/cooking/recipe/pie/impl/GardenPie.java
package content.global.skill.production.cooking.recipe.pie.impl;

import content.global.skill.production.cooking.recipe.pie.PieRecipe;
import core.game.node.item.Item;

/**
 * The Garden pie.
 */
public class GardenPie extends PieRecipe {

    private static final Item UNCOOKED_PIE = new Item(7176);

    private static final Item TOMATO = new Item(1982);

    private static final Item ONION = new Item(1957);

    private static final Item CABBAGE = new Item(1965);

    private static final Item PART_ONE = new Item(7172);

    private static final Item PART_TWO = new Item(7174);

    @Override
    public Item getProduct() {
        return UNCOOKED_PIE;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{TOMATO, ONION, CABBAGE};
    }

    @Override
    public Item[] getParts() {
        return new Item[]{PIE_SHELL, PART_ONE, PART_TWO, UNCOOKED_PIE};
    }

    @Override
    public boolean isSingular() {
        return false;
    }

}

File: Emulator/src/main/content/global/skill/cooking/recipe/pie/impl/GardenPie.kt
package content.global.skill.cooking.recipe.pie.impl

import content.global.skill.cooking.recipe.pie.PieRecipe
import org.rs.consts.Items
import core.game.node.item.Item

/**
 * Garden pie.
 */
class GardenPie : PieRecipe() {

    override fun getProduct(): Item {
        return UNCOOKED_PIE
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(TOMATO, ONION, CABBAGE)
    }

    override fun getParts(): Array<Item> {
        return arrayOf(PIE_SHELL, PART_ONE, PART_TWO, UNCOOKED_PIE)
    }

    override fun isSingular(): Boolean {
        return false
    }

    companion object {
        private val UNCOOKED_PIE = Item(Items.RAW_GARDEN_PIE_7176)
        private val TOMATO = Item(Items.TOMATO_1982)
        private val ONION = Item(Items.ONION_1957)
        private val CABBAGE = Item(Items.CABBAGE_1965)
        private val PART_ONE = Item(Items.PART_GARDEN_PIE_7172)
        private val PART_TWO = Item(Items.PART_GARDEN_PIE_7174)
    }
}
